GOVERNANCE.md

```diff
@@ -1,36 +1,87 @@
+
+> A large number of people requesting a fix or feature not being listened to for personal disagreements with the core maintenance team.
+
+> Somebody wanting to contribute a feature but not knowing where to begin, how to approach it, because it is not explained anywhere.
+
+> A pull-request lands which creates massive performance regression when compiled with `-O2`, affecting a large number of downstream users.
+
+> Losing adoption opportunities because the most popular options are too entrenched and are maintained by stubborn people.
+
+> A security flaw is discovered by researchers and we need to urgently upgrade a project, in a possibly breaking way.
+
+---
+
+This document elaborates on administrative and operational aspects of the Organisation, designed to serve and fullfill these goals as a priority.
 
 ## Administrative aspect
 
-The Organisation's administrative aspects are handled by the Core Team. While a healthy overlap is appreciated between the Core Team and the contributors,
-An employee of the Haskell Foundation has a position in the Core Team as to facilitate ecosystem-related matters and discussion with other institutional parties.
+The Organisation's administrative aspects are handled by the Core Team. The Core team are also contributors to projects.
+
+### Bylaws
+
+1. Core Team members are expected to do various kind of tasks (not niche things)
+2. Core Team members are expected to be available long-term, and it if it's not the case, they are to be substituted or replaced
+3. Core Team members may be required to do work when needed.
+4. The Core Team has no fixed number of seats, the number varies according to the needs of the Organisation.
+
+The non-Core Team contributors are not bound to these items. Non-Core Team contributors can make infrequent contributions, not maintain extended presence and not subject to working on-demand.
+
+### Selection process
+
+To be a part of the Core Team, the candidate needs to have a history of contributing to the projects hosted by the Organisation
+
+An Interim Core Team will be formed from an initial pool of volunteers.
+
+### Substitutions and Replacements
+
+Substitutions: Stepping-up for a fixed period of time when a Core Team member cannot fulfill their role;
+
+Replacement: Being a permanent replacement for a Core Team member;
 
 ## Operational aspect
 
 As an Organisation, `haskell-cryptography` concerns itself with two things:
 
-1. Code production
+1. Code production and maintenance
 2. Services related to the code and users
 
 ### Code production
 
 `haskell-cryptography` produces a series of projects related to cryptography.
-These projects have a referent person that is in charge of being the main point of contact, handle inquiries from the core team and act as an editor for the project.
+Each project has a key person that is in charge of being the main point of contact,
+handle inquiries from the core team and act as an editor for the project.
```

Comment:
Editor is as slightly confusing choice of words here; I assume it's meant in the sense of a newspaper editor, who isn't necessarily writing all the code, but is leading the direction of the project, provides feedback on submitted code etc.